### PR TITLE
(PA-533) Revert "Merge remote-tracking branch 'upstream/stable'"

### PR DESCRIPTION
**Pull request by McdonaldSeanp**
This reverts commit 6b1422f973bb17b85eabdca2e5b0ccbbe5ef2412, reversing
changes made to b733dab2d53a1708514ef30cf6e502e755081859.

This revert will roll back the ruby bump to 2.3.1

~~blocked on a merge of https://github.com/puppetlabs/puppet-agent/pull/853~~

### CODE REVIEW

**Comment by McdonaldSeanp** [comment]
Agent suite running at: https://jenkins.puppetlabs.com/view/puppet-agent/view/ad%20hoc/job/platform_puppet-agent_init-van-int_suite-manual-ad-hoc/337/

**Comment by McdonaldSeanp** [comment]
The suite seems to indicate everything is GTG for this PR.

**Comment by MosesMendoza** [comment]
Reviewing

File: README.md [hunk 1]
```diff
@@ -55,6 +56,31 @@ You also may need to change the source URIs for components. We recognize this is
 
 If you have your own mirror of the components of puppet-agent, you can also use a rewrite rule. See the [Vanagon README](https://github.com/puppetlabs/vanagon/blob/master/examples/projects/project.rb#L26) for an example.
 
+
+Building puppet-agent for windows
```
Comment (MosesMendoza) [inline README.md]: FYI it looks like this is reverting 424c8c6, which I think we want to keep eventually (updating readme to not reference puppet_for_the_win) - or does this revert mean we go back to using puppet_for_the_win, not vanagon for windows builds? I assume not. (this comment doesn't block merge)

**Comment by MosesMendoza** [comment]
I'm generally 👍  on this. It appears that this doesn't revert anything we _want_ to keep. I admit I have a hard time telling if it reverts _everything_ we don't want to keep, but not sure if that should block merge.

File: README.md (hunk 1 above)
Comment (McdonaldSeanp) [inline README.md]: No, we will not be going back to pftw, so yes we will need to put this PR back in.

**Comment by ScottGarman** [comment]
lgtm

**Pull request merged**